lib/dashboard.js

```
@@ -1,21 +1,81 @@
   process.env.SERVERLESS_PLATFORM_STAGE === 'dev'
     ? 'https://app.serverless-dev.com/'
     : 'https://app.serverless.com/';
 
-const getDashboardUrl = (ctx) => {
-  if (!ctx.sls.enterpriseEnabled) return dashboardUrl;
+const getServiceSpecificDashboardUrl = (ctx) => {
   const { service } = ctx.sls;
   return `${dashboardUrl}${service.org}/apps/${service.app}/${
     service.service
   }/${ctx.provider.getStage()}/${ctx.provider.getRegion()}`;
 };
 
-module.exports.getDashboardUrl = getDashboardUrl;
+// Left as-is as it is a part of public API used e.g. in Framework directly
+const getDashboardUrl = (ctx) => {
+  if (!ctx.sls.enterpriseEnabled) return dashboardUrl;
+  return getServiceSpecificDashboardUrl(ctx);
+};
 
-module.exports.getDashboardInteractUrl = (ctx) => {
+const getDashboardInteractUrl = (ctx) => {
   if (!ctx.sls.enterpriseEnabled) return null;
   return `${getDashboardUrl(ctx)}/interact`;
 };
+
+const dashboardHandler = async (ctx) => {
+  const { service } = ctx.sls;
+  const isServiceIntegratedWithDashboard = Boolean(service.org);
+
+  if (!isServiceIntegratedWithDashboard) {
+    process.stdout.write(
+      `This service does not use the Serverless Dashboard. Run ${chalk.bold(
+        'serverless'
+      )} to get started.\n`
+    );
+    return;
+  }
+
+  if (!isAuthenticated()) {
+    process.stdout.write(
+      `Could not find logged in user. Run ${chalk.bold('serverless login')} and try again.\n`
+    );
+    return;
+  }
+
+  let hasExistingDeployments = false;
+
+  try {
+    const platformSdk = await getPlatformClientWithAccessKey(service.org);
+    const deploymentsListResult = await platformSdk.frameworkDeployments.list({
+      orgName: service.org,
+      appName: service.app,
+      regionName: ctx.provider.getRegion(),
+      stageName: ctx.provider.getStage(),
+      serviceName: service.service,
+    });
+    if (deploymentsListResult.items.length) {
+      hasExistingDeployments = true;
+    }
+  } catch (e) {
+    // pass
```

This looks too permissive, e.g. any wrong call above will be silenced.

What exactly scenario we're supposed to ignore here?

---

we want to ignore any errors that might be thrown here really - it can throw if the app does not exist, it can throw due to network error, timeout, etc. What I really care about here is if I cannot determine if the service has deploys then I want to show simple link - I don't need any granular error handling here

What would you suggest then?

---

Ok. so it's purely about SDK communication errors. Let's then maybe introduce `checkForExistingDeployments` local function, which wraps with try/catch just those SDK calls, and logs eventually approached error if SLS_DEBUG is on.